lib/srv/app/watcher.go

```diff
@@ -82,7 +84,22 @@ func (s *Server) startResourceWatcher(ctx context.Context) (*services.AppWatcher
 		for {
 			select {
 			case apps := <-watcher.AppsC:
-				s.monitoredApps.setResources(apps)
+				appsWithAddr := make(types.Apps, 0, len(apps))
+				for _, app := range apps {
+					if app.GetPublicAddr() == "" {
+						c := app.Copy()
+						pubAddr, err := FindPublicAddr(s.c.AccessPoint, app.GetPublicAddr(), app.GetName())
+						if err == nil {
+							c.Spec.PublicAddr = pubAddr
+						} else {
+							s.log.WithError(err).Errorf("Unable to find public address for app %q, leaving empty.", app.GetName())
```

Comment:
Should we log it on error level if we ignore it? Maybe warn will be enough?

Reply:
There isn't much we can do with this error, but it is a genuine error and fixable by the Teleport admin. It is also rather an unlikely one to occur. All in all I think it should stay this way.